src/Analysis/Engine/Test/LineFormatterTests.cs

```diff
@@ -0,0 +1,460 @@
+﻿// Python Tools for Visual Studio
+// Copyright(c) Microsoft Corporation
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the License); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY
+// IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+//
+// See the Apache Version 2.0 License for specific language governing
+// permissions and limitations under the License.
+
+using System;
+using System.IO;
+using FluentAssertions;
+using Microsoft.Python.LanguageServer;
+using Microsoft.Python.LanguageServer.Implementation;
+using Microsoft.Python.Tests.Utilities.FluentAssertions;
+using Microsoft.PythonTools;
+using Microsoft.PythonTools.Analysis;
+using Microsoft.PythonTools.Analysis.FluentAssertions;
+using Microsoft.PythonTools.Parsing;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using TestUtilities;
+
+namespace AnalysisTests {
+    [TestClass]
+    public class LineFormatterTests {
+        [TestMethod, Priority(0)]
+        public void LineOutOfBounds() {
+            AssertNoEdits("a+b", line: 0);
+            AssertNoEdits("a+b", line: -1);
```

Comment:
With `DataTestMethodAttribute`, you can have parametrized tests which allows us to get separate result for each case.

Reply:
Jake may need an example :-)

Reply:
`AnalysisTest.PackageRelativeImportPep328`

Reply:
That looks really neat. I'll give it a go.

Reply:
Done.